pkg/util/httprequest/httprequest_test.go

```diff
@@ -171,6 +150,21 @@ func TestSchemeHost(t *testing.T) {
 				},
 			},
 			expectedScheme: "https",
+			expectedHost:   "route-namespace.router.default.svc.cluster.local",
```

Comment:
did you capture this via an actual request to haproxy and verify haproxy differentiates between an explicit port and an implicit one in the headers it sets?